superset/models/helpers.py

```diff
@@ -1348,6 +1348,7 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
         row_offset: Optional[int] = None,
         timeseries_limit: Optional[int] = None,
         timeseries_limit_metric: Optional[Metric] = None,
+        time_shift: Optional[str] = None,
```

Comment:
I'm not familiar with these helpers, but do we need to also copy over the new logic from `get_sqla_query()` in `connectors/sqla/models.py`? Or better yet, remove one of these to avoid code duplication?

Reply:
IMO, This helper has to do full refactoring. otherwise, all our code would have to be duplicated. the `time_shift` here is just to prevent raising an error from function calls.